doc/build-helpers/fixed-point-arguments.chapter.md

```diff
@@ -0,0 +1,89 @@
+# Fixed-point arguments of build helpers {#chap-build-helpers-finalAttrs}
+
+As mentioned in the beginning of this part, `stdenv.mkDerivation` could alternatively accept a fixed-point function. The input of such function, typically named `finalAttrs`, is expected to be the final state of the attribute set.
+A build helper like this is said to accept **fixed-point arguments**.
+
+As [`stdenv.mkDerivation`'s support of fixed-point arguments](#mkderivation-recursive-attributes) was first included in the Nixpkgs release 22.05, `mkDerivation`-style build helpers don't always come with such support yet.
+
+## Defining a build helper with `lib.extendMkDerivation` {#sec-build-helper-extendMkDerivation}
+
+The Nixpkgs library function [`lib.customisation.extendMkDerivation`](#function-library-lib.customisation.extendMkDerivation) can be used to define a build helper supporting fixed-point arguments from an existing one with such support. It can also bring fixed-point argument support to existing builders as a wrapper around the original definition.
+
+The following is the conventional definition of build helper `mkLocalDerivation` that doesn't support fixed-point arguments:
+
+:::{.example #ex-build-helpers-mkLocalDerivation-plain}
+
+# Example definition of `mkLocalDerivation` without fixed-point arguments support
+
+Write the followings in `pkgs/path/to/local-derivation/default.nix`:
+
+```nix
+{ lib
+, stdenv
+}:
+
+{ preferLocalBuild ? true
+, allowSubstitute ? false
+, impassablePredicate ? (_: false)
+, ...
+}@args:
+
+stdenv.mkDerivation (removeAttrs args [
+  "impassablePredicate"
+] // {
+  inherit
+    preferLocalBuild
+    allowSubstitute
+    ;
+})
+```
+
+and call the definition in [`all-packages.nix`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/all-packages.nix):
+
+```nix
+{
+  mkLocalDerivation = callPackage ../path/to/local-derivation { };
+}
+```
+
+:::
+
+A wrapper of `lib.extendMkDerivation` around the original definition brings the magical `finalAttrs` to the builder:
```

Comment:
> Kind of a nit-pick, but I don't think the documentation should encourage treating fix-points as “magical.” 

Thank you for pointing out! I do hope that such support could be a common practice for all `mkDerivatdon`-like build helpers.

> ```suggestion
> Wrapping the original definition with `lib.extendMkDerivation` requires making the inner function also accept the attrset's fixed-point, called `finalAttrs` here:
> ```

However, it doesn't impose such requirement for the resulting build helper (the whole definition evaluates to a build helper after being called by `callPackage`) to accept a fixed-point function instead of a plain attribute set. Plain attribute set as input argument is also supported, it only adds the the fixed-point argument as an option.

For the definition being wrapped, `finalAttrs` will always be available, but it doesn't mean one need to make additional changes to adopt it.

How about this?

```suggestion
Wrapping the original definition with `lib.extendMkDerivation` to make the result build helper also accept the attrset's fixed-point, called `finalAttrs` here:
```